### PR TITLE
[dbnode] allow replacing of seed node + docs

### PR DESCRIPTION
**Pull request by schallert**
Replacing a seed node requires some extra steps and was previously not
feasible using config alone. This adds a config flag to the seed node
list to allow a new seed node to join the etcd cluster, and adds docs
for replacing a seed node.

### CODE REVIEW

File: docs/operational_guide/placement_configuration.md [hunk 1]
```diff
@@ -154,3 +156,70 @@ curl -X POST <M3_COORDINATOR_HOST_NAME>:<M3_COORDINATOR_PORT(default 7201)>/api/
...
+  - hostID: host2
+    endpoint: http://1.1.1.2:2380
+  - hostID: host4
+    clusterState: existing
+    endpoint: http://1.1.1.4:2380
+```
+
+4. Start M3DB on `host4`.
+
+5. On all other seed nodes, update their `initialCluster` list to be exactly equal to the list on `host4` from step 3.
```
Comment (richardartoul) [inline docs/operational_guide/placement_configuration.md]: maybe obvious but should this also say: "Note, this means that you'll have to restart them one at a time" or something like that?

File: docs/operational_guide/placement_configuration.md [hunk 2]
```diff
@@ -154,3 +156,70 @@ curl -X POST <M3_COORDINATOR_HOST_NAME>:<M3_COORDINATOR_PORT(default 7201)>/api/
...
+
+6. Follow the steps from `Replacing a Seed Node` to replace `host3` with `host4` in the M3DB placement.
+
+Once the M3DB replace finishes, you may want to rolling restart the other seed nodes to ensure they have an up-to-date
```
Comment (richardartoul) [inline docs/operational_guide/placement_configuration.md]: ah ok you mention it here, might be better up above

File: docs/operational_guide/placement_configuration.md [hunk 3]
```diff
@@ -154,3 +156,70 @@ curl -X POST <M3_COORDINATOR_HOST_NAME>:<M3_COORDINATOR_PORT(default 7201)>/api/
     ]
 }'
 ```
+
+#### Replacing a Seed Node
```
Comment (richardartoul) [inline docs/operational_guide/placement_configuration.md]: Maybe add a disclaimer at the top of the file in bold saying something like: "If you're replacing an M3DB seed node please make sure to read the seed node specific instructions below."

File: docs/operational_guide/placement_configuration.md [hunk 4]
```diff
@@ -136,6 +136,8 @@ After sending the delete command you will need to wait for the M3DB cluster to r
 
 #### Replacing a Node
```
Comment (richardartoul) [inline docs/operational_guide/placement_configuration.md]: Can we add similar sections for add and remove? Basically in the add section say its not recommended to make new nodes seed nodes and if you find yourself in a situation where you need more than 3 M3DB seed nodes you should just run an external cluster and then something similar for remove?

File: docs/operational_guide/placement_configuration.md (hunk 3 above)
Comment (schallert) [reply docs/operational_guide/placement_configuration.md]: I mention this in `Replacing a Node`, we want it at the top as well?

Comment (richardartoul) [reply docs/operational_guide/placement_configuration.md]: Yeah sorry I mean more like: "If you're performing any placement operations on an M3DB seed node..." Just a general disclaimer incase they miss this specific heading

**Pull request merged**